addons/project/models/rating.py

```diff
@@ -0,0 +1,46 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+from odoo import api, fields, models
+
+
+class Rating(models.Model):
+    _inherit = 'rating.rating'
+
+    has_rating_access = fields.Boolean('Has Rating Access', compute='_compute_has_rating_access', search='_search_has_rating_access')
+
+    @api.depends('parent_ref')
+    def _compute_has_rating_access(self):
+        group_read = self.env['project.project'].read_group(
+            domain=['|', ('privacy_visibility', '!=', 'followers'),
+           '&', ('privacy_visibility', '=', 'followers'), ('message_is_follower')
+           ], fields=['id'], groupby=['id']
+        )
+        group_per_project_id = {group['id'][0]: group for group in group_read}
+        for rating in self:
+            group = group_per_project_id.get(rating.parent_res_id)
+            rating.has_rating_access = bool(group)
+
+    @api.model
+    def _search_has_rating_access(self, operator, value):
+        if not isinstance(value, bool):
+            raise ValueError('Invalid value: %s' % (value))
+        if operator not in ['=', '!=']:
+            raise ValueError('Invalid operator: %s' % (operator))
+
+        query = """
+            SELECT RR.id FROM rating_rating RR
+            JOIN project_project PP ON RR.parent_res_model = 'project.project' AND RR.parent_res_id = PP.id AND PP.privacy_visibility = 'followers'
+            JOIN mail_followers MF ON MF.res_model = 'project.project' AND MF.res_id = PP.id
+            WHERE MF.partner_id = %s GROUP BY RR.id
+
+            UNION
+
+            SELECT RR.id FROM rating_rating RR
+            JOIN project_project PP ON RR.parent_res_model = 'project.project' AND RR.parent_res_id = PP.id AND PP.privacy_visibility != 'followers'
+        """ % self.env.user.partner_id.id
+
+        if (operator == '=' and value is True) or (operator == '!=' and value is False):
+            operator_new = 'inselect'
+        else:
+            operator_new = 'not inselect'
+        return [('id', operator_new, (query, ()))]
```

Comment:
Indeed. Actually I don't know how people can write on ratings, except some specific controllers, as ratings should not be updated by hand (except some admins). Probably better remove write access completely and sudo in controllers / route / business methods once access is granted. Having that kind of computed fields always lead to issues and unexpected bugs.

Reply:
@xavierbol 

I agree with @tde-banana-odoo  that we can't write on ratings.
But here the case is different, 

Here we show reporting(Customer Ratings) menu to the project users to show their project's rating, In which users can see only those ratings for which they have access to that particular project(according to the project's privacy visibility).
So, to cover this case Either we can make a new reporting view(report model) that returns only their accessible ratings OR we have to add the compute field like above.

what do you prefer? Can you please suggest? Thanks

Reply:
Is it relevant to see "all my ratings" ? When you come from a project, doing an action with a domain is easy: no reporting, no field, ... just an action tailoring a domain. Could be simpler. And I bet seeing ratings in context of a given project is already a good start with less code ...

Reply:
@tde-banana-odoo 

Actually it is not for 'all ratings', It is for 'Customer Ratings' menuitem from reporting.

for ref: https://tinyurl.com/y2jfzufn
We want to show customer ratings menu to users. So, This is not only for project, but for all the projects which can be accessible by the user. So, if we pass domain for the action then we need to search from all the projects and this occurs permormance issue.

Could you please suggest?
cc: @xavierbol

Reply:
@pka-odoo 
If you agree with @tde-banana-odoo, then why you write in `rating.rating` model to do this (moreover, after my comment, you moved the code into `rating` module but the `parent_res_model` is not only the project one as I said in my previous comment)? 🤔

Indeed, I also agree with @tde-banana-odoo, it is not a good idea to write in rating model to solve the point in project. I discuss with our PO to avoid using the `Customer Ratings` menu for project user and use the "Customer Ratings" button in kanban card of project to see only the ratings for the project. 

![image](https://tinyurl.com/y5s94m3j)

Then, we could easily solve this point with just a domain in the action of this button, as Thibault proposes. By the way, we probably need to create a new action to add the domain only for this button and not in the "Customer Ratings" menu.

Reply:
I update my previous comment, the "Customer Ratings" button in the kanban view of `project.project` has the domain. So the project user could just use this button instead of allowing him to access to `Customer Ratings` menu in the `Reporting` one.

Reply:
@xavierbol Done.